core/src/main/java/com/amplifyframework/devmenu/DevMenuLogsFragment.java

```diff
@@ -54,11 +73,55 @@ public boolean onQueryTextSubmit(String searchText) {
 
             @Override
             public boolean onQueryTextChange(String searchText) {
-                logsText.setText(R.string.placeholder_logs);
-                logsText.setText(developerMenu.getFilteredLogs(searchText));
+                searchQuery = searchText;
+                displayFilteredLogs();
                 return true;
             }
         });
+        logLevelMenuId = R.id.all_logs;
+        Button filterLogs = logsView.findViewById(R.id.filter_logs);
+        registerForContextMenu(filterLogs);
+        filterLogs.setOnClickListener(view -> requireActivity().openContextMenu(view));
         return logsView;
     }
+
+    @Override
+    public void onCreateContextMenu(@NonNull ContextMenu menu, @NonNull View view,
+                                    @Nullable ContextMenu.ContextMenuInfo menuInfo) {
+        super.onCreateContextMenu(menu, view, menuInfo);
+        requireActivity().getMenuInflater().inflate(R.menu.dev_menu_logs_menu, menu);
+        menu.findItem(logLevelMenuId).setChecked(true);
+    }
+
+    @Override
+    public boolean onContextItemSelected(MenuItem item) {
+        int itemId = item.getItemId();
+        if (itemId == R.id.all_logs) {
```

Comment:
This is a perfect place to use a Switch statement instead of the if else chain.

Reply:
Also, I wonder if there's a way you can simply include the LogLevel value in each MenuItem.

Reply:
I did try to use a Switch statement instead of all the if else's, but resource IDs cannot be used in switch statements.